whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -112,50 +115,69 @@ protected SearchResponse doSearch() throws IOException {
                 final SearchHit[] hits = fulltextResponse.getHits().getHits();
 
                 LOGGER.debug("ElasticSearch {} hits for the query: {}", hits.length, searchRequest.getQuery());
-                final List<RpslObject> results = Lists.newArrayList();
-                int resultSize = Math.min(maxResultSize,Long.valueOf(fulltextResponse.getHits().getTotalHits().value).intValue());
+                return prepareResponse(fulltextResponse, hits, searchRequest, stopwatch);
+            }
+        }.search();
+    }
+
+    private SearchResponse prepareResponse(final org.elasticsearch.action.search.SearchResponse fulltextResponse,
+                                           final SearchHit[] hits,
+                                           final SearchRequest searchRequest,
+                                           final Stopwatch stopwatch) {
+        final List<SearchResponse.Lst> highlightDocs = Lists.newArrayList();
+        final List<SearchResponse.Result.Doc> resultDocumentList = Lists.newArrayList();
+        fillHighlightsAndResults(hits, highlightDocs, resultDocumentList);
 
-                final SearchResponse.Lst highlight = new SearchResponse.Lst("highlighting");
-                final List<SearchResponse.Lst> highlightDocs = Lists.newArrayList();
+        final SearchResponse.Result result = new SearchResponse.Result("response", Long.valueOf(fulltextResponse.getHits().getTotalHits().value).intValue(), searchRequest.getStart());
+        result.setDocs(resultDocumentList);
+
+        final SearchResponse searchResponse = new SearchResponse();
+        searchResponse.setResult(result);
 
-                for (final SearchHit hit : hits) {
-                    try {
-                        final RpslObject rpslObject = objectDao.getById(Integer.parseInt(hit.getId()));
-                        results.add(rpslObject);
-                        account(rpslObject);
+        final List<SearchResponse.Lst> responseLstList = Lists.newArrayList();
```

Comment:
You removed the account method this method counts the personal objects limit and throws exception in case user breaches the limit. Where are you accounting for the personal objects now?  I think there is a test case also if not ca you add one?

Reply:
pushed using the account